### DISCUSSION TITLE
Merge commits are not allowed on this repository.

### DISCUSSION THREAD

**Discussion by guguducken** [Mergifyio/mergify#5086] [Q&A]
When I use mergify queue to merge my pull request to main branch, Mergify had not worked and showed this: 
> Merge commits are not allowed on this repository.
The repository configuration doesn't allow merge commits. The merge method configured in Mergify configuration must be allowed in the repository configuration settings.

This is why? It worked fine before

**Comment by jd** [comment]
Hi @guguducken 

Could you share your configuration URL or content?
The error seems to indicate that your repository is configured to disallow merge commits but your `method` for merging is `merge` (the default).

**Reply by guguducken** [reply]
this: https://github.com/matrixorigin/matrixone/blob/main/.github/mergify.yml
but the default method in mergify is `squash`

**Reply by jd** [reply]
Great, thank you. Can you share the PR URL where you saw that error? That'd help finding out the root cause.

**Comment by naancy0245** [comment]
"Merge commits are not allowed on this repository" indicates that the repository in question has a specific policy in place to disallow merge commits. This could be due to various reasons, such as ensuring that code changes are made through pull requests to maintain code quality and consistency. To resolve this error, one would need to follow the repository's guidelines and make code changes without merging commits or reaching out to the repository maintainers for clarification on their policy.
